src/sidebar/components/select-filter.js

```diff
@@ -0,0 +1,69 @@
+import { createElement } from 'preact';
+import propTypes from 'prop-types';
+
+import Menu from './menu';
+import MenuItem from './menu-item';
+import SvgIcon from '../../shared/components/svg-icon';
+
+/**
+ * @typedef {import('../store/modules/filters').FilterOption} FilterOption
+ */
+
+/**
+ * @typedef SelectFilterProps
+ * @prop {FilterOption} defaultOption
+ * @prop {string} [icon]
+ * @prop {(selectedFilter: FilterOption) => any} onSelect
+ * @prop {FilterOption[]} options
+ * @prop {FilterOption} [selectedOption]
+ * @prop {string} title
+ */
+
+/**
+ * A select-element-like control for selecting one of a defined set of
+ * options.
+ *
+ * @param {SelectFilterProps} props
+ */
+function SelectFilter({
+  defaultOption,
+  icon,
+  onSelect,
+  options,
+  selectedOption,
+  title,
+}) {
+  const filterOptions = [defaultOption, ...options];
+  const selected = selectedOption ?? defaultOption;
+
+  const menuLabel = (
+    <span className="select-filter__menu-label">
+      {icon && <SvgIcon name={icon} className="select-filter__menu-icon" />}
+      {selected.display}
+    </span>
+  );
+
+  return (
+    <Menu label={menuLabel} title={title}>
```

Comment:
One thing this PR lacks is a11y tests. It's on my list. I may submit a separate PR with just that. I want to be absolutely certain that using a Menu component here is accessible. Again, by encapsulating this in a single `SelectFilter` component, I'm hoping it will be easy to swap out if we decide against it.